flake.nix

```diff
@@ -22,10 +29,10 @@
 
       devShell.x86_64-linux = import ./devShell.nix { system = "x86_64-linux"; };
 
-      lib = flake-utils.lib // {
-        # modulesFromList is deprecated, will be removed in future releases
-        inherit systemFlake modulesFromList;
+      lib = flake-utils.lib // internal-functions // {
```

Comment:
I would advise against exporting all of `internal-functions`: it puts unnecessary extra burden on backwards compatibility and deprecation requirements. It's a well-known truth that people inevitable start to depend on public APIs, so let's be careful about it.

I would only export the main functions of this library. Since `rakeLEaves` is now included here (:+1:), I'd also export this specifically, since we make heavy use in devos flakes.

Another function that's useful outside of `fup` is the one that reverses a hostname into a reverse dns notation. I already found myself re-implementing this in various versions downstream.

Reply:
These internal functions can either be in namespace and be like "if you are using it - your problem" or each file can import it manually. Or at least these are solutions I see

Reply:
I will export the main functions in the top-level lib, and if you're alright with it, add `lib.internal` which contains everything inside.

Reply:
By the way @blaggacao how do you suggest I add the `reverseDns` function? I feel like it's a bit scattered in `mkFlake`. Should I combine `domain`, `dnsLabels` and `domain`?

Maybe there's some more to the functionality which I haven't discovered yet.

Reply:
@fufexan Now, that I look at it again, maybe we should leave out the reverse dns thing. It's already too bound to the nixos networking module.